Tests/Common/Service/HttpEndpointTest.swift

```diff
@@ -44,4 +44,29 @@ class HttpEndpointTest: UnitTest {
 
         XCTAssertEqual(actual, expected)
     }
+
+    func test_getUrlString_givenRandomPathNeedsEncoding_expectEncodedPath() {
+        let identifierWithSpecialChar = String.randomStringWithSpecialCharacters()
```

Comment:
I would prefer if we hard-coded test data instead of randomly generate it. Maybe the randomly generated string gives us a char with no special chars, for example. 

The other test function is great in that it includes a special character that we have identified causes issues for customers. I like the idea of hard-coding test strings that we know could cause an issue.

Reply:
Good point, but random values does extend range for testing more characters. We can perhaps update `randomStringWithSpecialCharacters` to must include special character every time.